app/controllers/books_controller.rb

```diff
@@ -15,14 +15,38 @@ def create
     @book = Book.new(book_params)
 
     if @book.save
-      redirect_to :action => 'index'
+      redirect_to action: 'index'
     else
-      @subjects = Subject.all
-      render :action => 'new'
+      render action: 'new'
     end
   end
 
   def book_params
-    params.require(:books).permit(:title, :author, :isbn)
+    params.require(:book).permit(:title, :author, :isbn)
+  end
+
+  def edit
+    @book = Book.find(params[:id])
+  end
+
+  def update
+    @book = Book.find(params[:id])
+
+    if @book.update_attributes(book_param)
+      redirect_to action: 'index'
+    else
+      render action: 'edit'
+    end
+  end
+
+  def book_param
+    params.require(:book).permit(:title, :price, :subject_id, :description)
+  end
+
+  # I was expecting to call this 'delete' since that is what 'method' calls
+  # Why is it destroy instead? Is there any way I could have guessed that?
```

Comment:
The documentation mentions it: https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions -- I believe the reason is that Rails specifically wants different names from the HTTP verbs.